### PR TITLE
feat: clap wrapper around sqlness 

### PR DESCRIPTION
**Pull request by Lilit0x**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Added clap to the `sqlness` runner to improve the interface by providing help messages and allowing configurations.

Please explain IN DETAIL what the changes are in this PR and why they are needed:
`sqlness` can now accept arguments to be configured with when running tests. Below is the result of the help command after running: `cargo sqlness -h`
```
SQL Harness for GrepTimeDB

Usage: sqlness-runner [OPTIONS]

Options:
  -c, --case-dir <CASE_DIR>                Directory of test cases
  -f, --fail-fast...                       Fail this run as soon as one case fails if true
  -e, --env-config-file <ENV_CONFIG_FILE>  Environment Configuration File [default: config.toml]
  -t, --test-filter <TEST_FILTER>          Name of test case to be matched [default: ]
  -h, --help                               Print help
  -V, --version                            Print version
```
All arguments assume their default values when omitted from the command line. To use all arguments(or some):
```sh
cargo sqlness -f -c /path/to/testcase -e example.toml -t aggregation
```

## Checklist
- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

This PR references #2380

### CODE REVIEW

File: tests/runner/src/main.rs [hunk 1]
```diff
@@ -12,31 +12,51 @@
...
+struct Args {
+    /// Directory of test cases
+    #[clap(short, long)]
+    case_dir: Option<PathBuf>,
+
+    ///Fail this run as soon as one case fails if true
+    #[arg(short, long, action = clap::ArgAction::Count)]
+    fail_fast: u8,
+
+    ///Environment Configuration File
```
Comment (waynexia) [inline tests/runner/src/main.rs:36]: nit-picking:
```suggestion
    /// Environment Configuration File
```

File: tests/runner/src/main.rs [hunk 2]
```diff
@@ -12,31 +12,51 @@
...
+    #[clap(short, long, default_value = "config.toml")]
+    env_config_file: String,
+
+    ///Name of test case to be matched
```
Comment (waynexia) [inline tests/runner/src/main.rs:40]:
```suggestion
    /// Name of test cases to run. Accept as a regexp.
```

File: tests/runner/src/main.rs [hunk 3]
```diff
@@ -12,31 +12,51 @@
...
+#[clap(author, version, about, long_about = None)]
+/// SQL Harness for GrepTimeDB
+struct Args {
+    /// Directory of test cases
+    #[clap(short, long)]
+    case_dir: Option<PathBuf>,
+
+    ///Fail this run as soon as one case fails if true
+    #[arg(short, long, action = clap::ArgAction::Count)]
+    fail_fast: u8,
```
Comment (waynexia) [inline tests/runner/src/main.rs:34]: Prefer to declare this as `bool`. Default to not fail fast, and if this option is given then switch to fail-fast mode

File: tests/runner/src/main.rs [hunk 4]
```diff
@@ -12,31 +12,51 @@
...
 
+#[derive(Parser, Debug)]
+#[clap(author, version, about, long_about = None)]
+/// SQL Harness for GrepTimeDB
+struct Args {
+    /// Directory of test cases
+    #[clap(short, long)]
+    case_dir: Option<PathBuf>,
+
+    ///Fail this run as soon as one case fails if true
```
Comment (waynexia) [inline tests/runner/src/main.rs:32]:
```suggestion
    /// Fail this run as soon as one case fails if true
```

File: tests/runner/src/main.rs [hunk 5]
```diff
@@ -12,31 +12,51 @@
...
+    #[arg(short, long, action = clap::ArgAction::Count)]
+    fail_fast: u8,
+
+    ///Environment Configuration File
+    #[clap(short, long, default_value = "config.toml")]
+    env_config_file: String,
+
+    ///Name of test case to be matched
+    #[clap(short, long, default_value = "")]
```
Comment (waynexia) [inline tests/runner/src/main.rs:41]: According to the sqlness doc, the default value should be `.*`

**Review by waynexia — APPROVED** [review]
LGTM, appreciate it ❤️

**Review by evenyag — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**